packages/core/schematics/migrations/ngclass-to-class-migration/ngclass-to-class-migration.spec.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {absoluteFrom} from '@angular/compiler-cli';
+import {diffText, runTsurgeMigration} from '../../utils/tsurge/testing';
+import {NgClassMigration} from './ngclass-to-class-migration';
+import {initMockFileSystem} from '@angular/compiler-cli/src/ngtsc/file_system/testing';
+
+describe('ngClass migrator', () => {
+  beforeEach(() => {
+    initMockFileSystem('Native');
+  });
+
+  describe('No change cases', () => {
+    it('should not change static HTML elements', async () => {
+      await verifyDeclarationNoChange(`<button id="123"></button>`);
+    });
+
+    it('should not change existing [class] bindings', async () => {
+      await verifyDeclarationNoChange(`<div [class.active]="isActive"></div>`);
+    });
+
+    it('should not change empty ngClass binding', async () => {
+      await verifyDeclarationNoChange(`<div [ngClass]="{}"></div>`);
+    });
+
+    it('should not change ngClass with empty string key', async () => {
+      await verifyDeclarationNoChange(`<div [ngClass]="{'': condition}"></div>`);
+    });
+
+    it('should not change ngClass with empty array', async () => {
+      await verifyDeclarationNoChange(`<div [ngClass]="[]"></div>`);
+    });
+  });
+
+  describe('Simple ngClass object migrations', () => {
+    it('should migrate single condition', async () => {
+      await verifyDeclaration({
+        before: `<div [ngClass]="{'active': isActive}"></div>`,
+        after: `<div [class.active]="isActive"></div>`,
+      });
+    });
+
+    it('should migrate multiple class conditions', async () => {
+      await verifyDeclaration({
+        before: `<div [ngClass]="{'class1': condition1, 'class2': condition2}"></div>`,
+        after: `<div [class.class1]="condition1" [class.class2]="condition2"></div>`,
```

Comment:
Class bindings to object are valid. (as long as there is no space in the keys/keys are not dynamic)

```suggestion
        before: `<div [ngClass]="{'class1': condition1, 'class2': condition2}"></div>`,
        after: `<div [class]="{'class1': condition1, 'class2': condition2}"></div>`,
```